### PR TITLE
[Messenger] Fix PHPDoc

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #... 
| License       | MIT

https://psalm.dev/docs/annotating_code/type_syntax/conditional_types/
https://psalm.dev/r/76330fe331

### CODE REVIEW

File: src/Symfony/Component/Messenger/Envelope.php [hunk 1]
```diff
@@ -110,9 +110,7 @@ public function last(string $stampFqcn): ?StampInterface
      *
      * @param class-string<TStamp>|null $stampFqcn
      *
-     * @return StampInterface[]|StampInterface[][] The stamps for the specified FQCN, or all stamps by their class name
-     *
-     * @psalm-return ($stampFqcn is string : array<class-string<StampInterface>, list<StampInterface>> ? list<TStamp>)
+     * @return ($stampFqcn is string ? list<TStamp> : array<class-string<StampInterface>, list<StampInterface>>) The stamps for the specified FQCN, or all stamps by their class name
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/Envelope.php:113]: why merge return and psalm-return? I think we didn't anywhere yet so we shouldn't start in a PR like that

Comment (ro0NL) [reply src/Symfony/Component/Messenger/Envelope.php:113]: i thought it made psalm trip, but no, it still errors. We should template stamps property still, but that's not a public API concern. Also i was unsure how to do it :sweat_smile: 

in fact we use both conditional `@return` and `@psalm-return`. Merge across the board welcome i think.

https://github.com/search?q=repo%3Asymfony%2Fsymfony+%22%40return+%28%22&type=code
https://github.com/search?q=repo%3Asymfony%2Fsymfony+%22%40psalm-return+%28%22&type=code

Comment (nicolas-grekas) [reply src/Symfony/Component/Messenger/Envelope.php:113]: I don't think we use conditional types in `@return`

Comment (ro0NL) [reply src/Symfony/Component/Messenger/Envelope.php:113]: we do https://github.com/search?q=repo%3Asymfony%2Fsymfony+%22%40return+%28%22&type=code

Comment (ro0NL) [reply src/Symfony/Component/Messenger/Envelope.php:113]: we do https://github.com/search?q=repo%3Asymfony%2Fsymfony+%22%40return+%28%22&type=code

Comment (ro0NL) [reply src/Symfony/Component/Messenger/Envelope.php:113]: i dont mind, you can cherry pick a2c660b94410f4abf54e79b932702a3172ce8a9f

**Pull request closed without merging**